File: javapatterns/src/main/java/org/iut/nantes/adapter/chimie/DBManager.java
package org.iut.nantes.adapter.chimie;

import java.util.ArrayList;

class DBManager
{
  private OldDB ancienneBD_;

  /**** code ajouté ****/
  private NewDB nouvelleBD_;
  private AdapterDB adapter_;
  /*********************/

  private ArrayList<OldDB> BDs;
  
  public DBManager()
  {
    Molecule eau = new Molecule( "eau", "H2O" );
    Molecule ethanol = new Molecule( "ethanol", "CH3-CH2-OH" );
    
    ancienneBD_ = new OldDB();
    ancienneBD_.oldAdd( eau );

    /**** code ajouté ****/
    nouvelleBD_ = new NewDB();
    nouvelleBD_.newAdd( ethanol );

    adapter_ = new AdapterDB( nouvelleBD_ );
    /*********************/

    BDs = new ArrayList<OldDB>();
    BDs.add( ancienneBD_ );

    /**** code ajouté ****/
    BDs.add( adapter_ );
    /*********************/
  }

  public void printMolecule( String name )
  {
    boolean printSomething = false;
    
    for( OldDB bd : BDs )
      if( bd.exists( name ) )
      {
	bd.oldMolecule( name ).affiche();
	printSomething = true;
	break;
      }
    
    if( !printSomething )
      System.out.println( "Molecule inconnue." );
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/adapter/chimie/DBManager.kt
package org.iut.nantes.adapter.chimie

import java.util.*

internal class DBManager {
    private val ancienneBD_: OldDB

    /**** code ajouté  */
    private val nouvelleBD_: NewDB
    private val adapter_: AdapterDB

    /** */
    private val BDs: ArrayList<OldDB>

    init {
        val eau = Molecule("eau", "H2O")
        val ethanol = Molecule("ethanol", "CH3-CH2-OH")
        ancienneBD_ = OldDB()
        ancienneBD_.oldAdd(eau)
        /**** code ajouté  */
        nouvelleBD_ = NewDB()
        nouvelleBD_.newAdd(ethanol)
        adapter_ = AdapterDB(nouvelleBD_)
        /** */
        BDs = ArrayList()
        BDs.add(ancienneBD_)
        /**** code ajouté  */
        BDs.add(adapter_)
        /** */
    }

    fun printMolecule(name: String?) {
        var printSomething = false
        for (bd in BDs) if (bd.exists(name)) {
            bd.oldMolecule(name)!!.affiche()
            printSomething = true
            break
        }
        if (!printSomething) println("Molecule inconnue.")
    }


}